OADP-4655 Added section for kopia env variables (#80869)
openshift/openshift-docs: modules/oadp-usecase-kopia-override-algorithms.adoc

// Module included in the following assemblies:
//
// * backup_and_restore/application_backup_and_restore/installing/overriding-kopia-algorithms.adoc          

:_mod-docs-content-type: PROCEDURE
[id="oadp-usecase-kopia-override-algorithms.adoc_{context}"]
= Use case for overriding Kopia hashing, encryption, and splitter algorithms

> This use case walks you through the steps for taking a backup with Kopia environment variables for hashing, encryption, and splitter. You store the backup in an AWS S3 bucket. You then verify the environment variables by connecting to the Kopia repository.

Reviewer: "This might be a vague referent" and "walks" might be confusing to "non-native-English" speakers. How about:

> The use case example demonstrates making a backup <of an application?> by using Kopia environment variables for hashing, encryption, and splitter tasks.

Author: Done.

.Prerequisites

* You have installed the {oadp-short} Operator.
* You have an AWS S3 bucket configured as the backup storage location.
* You have created the secret by using the credentials provided by the cloud provider.

> * You have installed the Kopia client.

Reviewer 2: Not required for using it, required for verification part.

Author: I am keeping this one as we have a 'Verification' section at the end and user runs the kopia command to verify.

* You have an application with persistent volumes running in a separate namespace.

.Procedure

> . Configure the DPA as shown in the following example:

Reviewer: Spell out "DPA" on first use in a module in the event the model is somehow reused elsewhere? Each module should be independent of other modules.

Author: done.

+
[source,yaml]
----
apiVersion: oadp.openshift.io/v1alpha1
kind: DataProtectionApplication
metadata:
name: <dpa_name> # <1>
namespace: openshift-adp
spec:
backupLocations:
- name: aws
  velero:
    config:
      profile: default
      region: <region_name> # <2>
    credential:
      key: cloud
      name: cloud-credentials # <3>
    default: true
    objectStorage:
      bucket: <bucket_name> # <4>
      prefix: velero
    provider: aws
configuration:
  nodeAgent:
    enable: true
    uploaderType: kopia
  velero:
    defaultPlugins:
    - openshift
    - aws
    - csi # <5>
    defaultSnapshotMoveData: true
    podConfig:
      env:
        - name: KOPIA_HASHING_ALGORITHM
          value: BLAKE3-256 # <6>
        - name: KOPIA_ENCRYPTION_ALGORITHM
          value: CHACHA20-POLY1305-HMAC-SHA256 # <7>
        - name: KOPIA_SPLITTER_ALGORITHM

>           value: DYNAMIC-8M-RABINKARP # <8>

Reviewer: ```suggestion
          value: DYNAMIC-8M-RABINKARP # <8>
# ...
```

Author: The yaml is fully complete. I have myself tested the yaml on a QE cluster. Therefore, not including the '...'. Thanks.

  name: test-backup
  namespace: openshift-adp
spec:
  includedNamespaces:
  - <application_namespace> # <1>
  defaultVolumesToFsBackup: true
----
<1> Specify the namespace for the application installed in the cluster.

. Create a backup by running the following command:
+
[source,terminal]
----
$ oc apply -f <backup_file_name> # <1>
----
<1> Specify the name of the backup CR file.
+
. Verify that the backup completed by running the following command:
+
[source,terminal]
----
$ oc get backup <backup_name> -o yaml # <1>
----
<1> Specify the name of the backup.

.Verification
. Connect to the Kopia repository by running the following command:
+
[source,terminal]
----
$ kopia repository connect s3 \
  --bucket=<bucket_name> \ # <1>
  --prefix=velero/kopia/<application_namespace> \ # <2>
  --password=static-passw0rd \ # <3>
  --access-key="<aws_s3_access_key>" \ # <4>
  --secret-access-key="<aws_s3_secret_access_key>" \ # <5>
----
<1> Specify the AWS S3 bucket name.
<2> Specify the namespace for the application.
<3> This is the Kopia password to connect to the repository.
<4> Specify the AWS S3 access key.
<5> Specify the AWS S3 storage provider secret access key.
+
[NOTE]
====

> If you are using a storage provider other than AWS S3, you will need to add `--endpoint` , the bucket endpoint URL parameter, in the command.

Reviewer: ```suggestion
If you are using a storage provider other than AWS S3, you will need to add `--endpoint`, the bucket endpoint URL parameter, to the command.
```

====
+
+

> . Verify that Kopia uses the environment variables configured in the DPA for the backup. Run the following command:

Reviewer: ```suggestion
. Verify that Kopia uses the environment variables that are configured in the DPA for the backup by entering the following command:
```
